### DISCUSSION TITLE
Multiple select tags, on change, pass all values

### DISCUSSION THREAD

**Discussion by jritsema** [bigskysoftware/htmx#1554] [Q&A] [answered]
Hi. I have multiple `<select>` tags used as filters.  When the user changes any of them, I would like HTMX to issue an AJAX GET request sending all `<select>` values as query string parameters (or I suppose POST values should work too).  Is this possible with HTMX?

**Comment by Telroshan** [comment, marked as answer]
Hello, check out [this JSFiddle](https://jsfiddle.net/r31t7bym/) for an example implementation.
```html
<select name="filter1">
  <option value="1">1</option>
  <option value="2">2</option>
</select>
<select name="filter2">
  <option value="A">A</option>
  <option value="B">B</option>
</select>


<div hx-get="/url" hx-include="select" hx-trigger="change from:select">
 Content will be swapped here
</div>
```
- Use [hx-get](https://htmx.org/attributes/hx-get/) to issue GET requests. Similarly, use [hx-post](https://htmx.org/attributes/hx-post/) to issue a POST request instead
- Use [hx-include](https://htmx.org/attributes/hx-include/) when you can't use a form or need to include elements that are elsewhere in the DOM hierarchy, to include the required values
- Use [hx-trigger](https://htmx.org/attributes/hx-trigger/) to specify when the request should be sent. Here, I used the standard [`change`](https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/change_event) event, along with the `from:selector` specifier for hx-trigger that htmx provides _(cf doc)_

Please note that there are several ways of accomplishing the same result, depending on your situation, your constraints, and your preferences
- You could use a form to wrap the selects, define hx-get on it, and skip the hx-include attribute _(as values inside a form are automatically included along)_
- You could set hx-get on every select element and set hx-trigger to `hx-trigger="change"` instead of using the `from` syntax
- Very likely other ways that I just can't think of from the top of my head as I'm writing this

Hope this helps!

**Comment by jritsema** [comment]
Thank you @Telroshan!  The `"change from:select"` was what I was missing.  Didn't realize you could combine them :)
